### DISCUSSION TITLE
Ability to specify if subproject should be cross-compiled or not

### DISCUSSION THREAD

**Discussion by LunarLambda** [mesonbuild/meson#11120] [Q&A] [answered]
Project A can be cross-compiled (it's a library for command-line parsing), so does not specify `native: true`.

Project B must be cross-compiled (single-platform embedded project), but wants to use Project A as a subproject dependency for `native: true` executables.

Is there any way for Project B to tell meson it should build Project A using the native compiler?

Or would Project A need to expose a custom build option (i.e. `-Dnative=true`), or would it need to expose two dependencies (one cross, one native)

Something like `subproject(..., native: true)` would be nice, however I can see this clashing with the rule of "every subproject is executed only once", as well as fallback subprojects used with wraps.

**Comment by dcbaker** [comment, marked as answer]
This is a known issue, that subprojects are always built for the host machine, and never for the build machine. Unfortunately fixing it is *extremely* non-trivial (I've looked at it before). This is something I think we want to fix, but it's a big project with some non-obvious implications.

I've created an issue for this (since we don't seem to have one yet)

**Reply by xclaesse** [reply]
It would be really great, but non trivial at all. One great side effect would be to be able to build a subproject for both machines, which often helps when you need to build some tools for build machine and a library for host machine.

**Reply by LunarLambda** [reply]
I'll stick with my build option plan for now, then. Glad to hear it's already something on people's minds.

**Comment by jhgorse** [comment]
For documentation's sake, how do we do this now? Run `meson setup` twice, with and without --cross-file?
